js-task4-object.html

```diff
@@ -9,51 +9,57 @@
   <script>
     'use strict';
     {
+      // タスク一覧の見出し
       const taskText = [
         { text: '======================\n現在待っているタスク一覧\n======================' },
       ];
-      const taskList = [];
-      const task = {
-        task: '机を片付ける',
-        genre: '掃除',
-      };
-      const task1 = {
-        task: '牛乳を買う',
-        genre: '買い物',
-      };
-      const task2 = {
-        task: '散歩する',
-        genre: '運動',
-      };
-      taskList.push(task);
-      taskList.push(task1);
-      taskList.push(task2);
+      // 既存のタスクを追加
+      function tasksList() {
+        const taskList = [];
+        const task = {
+          task: '机を片付ける',
+          genre: '掃除',
+        };
+        const task1 = {
+          task: '牛乳を買う',
+          genre: '買い物',
+        };
+        const task2 = {
+          task: '散歩する',
+          genre: '運動',
+        };
+        taskList.push(task);
+        taskList.push(task1);
+        taskList.push(task2);
+        return taskList;
+      }
+      tasksList();
+      const taskList = tasksList();
 
-      function showList() {
+      //taskListを繰り返し処理で表示させる
+      function showList(taskList) {
+        let n = taskList.length;
         console.log(taskText[0].text);
-        for (let i = 0; i < 3; i++) {
+        for (let i = 0; i < n; i++) {
```

Comment:
```suggestion
        for (let i = 0; i < taskList.length; i++) {
```
とする方が一般的かつ、
この前の行で宣言されている変数 `n` が以降の行で全く使われてないようなので、
変数 `n` の宣言が冗長に感じます。